### ISSUE TITLE
Parse error when parsing plpgsql function with cursor

### ISSUE THREAD

**Issue by svenklemm** [lelit/pglast#154]
To reproduce:

```
>>> sql = """
...    CREATE FUNCTION foo(cmd TEXT) RETURNS void AS $$
...    DECLARE
...      i INT;
...      c CURSOR FOR SELECT generate_series(1,10);
...    BEGIN
...      FOR i IN c LOOP
...        RAISE NOTICE 'i is %',i;
...      END LOOP;
...    END
...    $$ LANGUAGE plpgsql;
... """
>>> pglast.parse_plpgsql(sql)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sven/pgspot/lib/python3.12/site-packages/pglast/__init__.py", line 38, in parse_plpgsql
    return loads(parse_plpgsql_json(statement))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pglast/parser.pyx", line 348, in pglast.parser.parse_plpgsql_json
pglast.parser.ParseError: syntax error at or near "c"
>>> 
```

libpg_query seems to have no problem parsing that function

**Comment by lelit** [comment]
Thanks for the report.

In general, the _parser_ functions exposed by `pglast` are just tiny wrappers around corresponding functions in `libpg_query`: for this reason, `ParseError` are very likely due to either a real syntax issue, or a problem in the underlying `libpg_query` library.

This seems the latter case, since transposing your example into C gives the same error:
```c
#include <pg_query.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
  PgQueryPlpgsqlParseResult result;

  result = pg_query_parse_plpgsql("\
   CREATE FUNCTION foo(cmd TEXT) RETURNS void AS $$\
   DECLARE\
     i INT;\
     c CURSOR FOR SELECT generate_series(1,10);\
   BEGIN\
     FOR i IN c LOOP\
       RAISE NOTICE 'i is %',i;\
     END LOOP;\
   END\
   $$ LANGUAGE plpgsql;");
  if (result.error) {
    printf("error: %s at %d\n", result.error->message, result.error->cursorpos);
  } else {
    printf("%s\n", result.plpgsql_funcs);
  }

  pg_query_free_plpgsql_parse_result(result);

  // Optional, this ensures all memory is freed upon program exit (useful when running Valgrind)
  pg_query_exit();

  return 0;
}
```

gives
```sh
cc -g -I. -I./vendor -Wall -o examples/simple_plpgsql -g examples/simple_plpgsql.c libpg_query.a -pthread
examples/simple_plpgsql 
error: syntax error at or near "c" at 0
```

I will forward this to the `libpg_query` team.

**Comment by svenklemm** [comment]
I tried with libpq_query and it succeeded maybe pglast just needs a version bump

**Comment by svenklemm** [comment]
Which version of libpg_query did you test this against?

**Comment by svenklemm** [comment]
Oh actually in 16-latest it fails but in my own branch it works. 
https://github.com/pganalyze/libpg_query/pull/256

**Comment by lelit** [comment]
Thank you for additional details: I usually refrain using non-merged forks of libpg_query, because that typically heads to confusion/complication. Is it reasonable for you to wait until that happens?

**Comment by svenklemm** [comment]
Sure, thats fine. Hopefully the PR gets merged soon

**Comment by svenklemm** [comment]
The fix has been merged into libpg_query

**Comment by lelit** [comment]
Right, I noticed: although it wasn't properly released, I just pushed v6.4 targeting its `16-latest` branch.
Thanks for solving this!
